crates/hotshot/src/lib.rs

```diff
@@ -740,7 +639,7 @@ pub struct HotShotConsensusApi<TYPES: NodeType, I: NodeImplementation<TYPES>> {
 }
 
 #[async_trait]
-impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusSharedApi<TYPES, I>
+impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusApi<TYPES, I>
```

Comment:
I noticed in the PR description you mentioned that we could not remove `ConsensusApi` directly because some functions were still used--good catch! However, I'm wondering if it's indeed necessary to have this trait. Most of its functions are just getters, which I think makes more sense to be implemented in `HotShotConsensusApi`, especially now that only `HotShotConsensusApi` uses these functions.

Could we move those needed functions to `HotShotConsensusApi` and delete unused ones?